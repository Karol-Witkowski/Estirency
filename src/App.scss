@import url('https://fonts.googleapis.com/css?family=Josefin+Sans:200,300,400,500,600,700&display=swap');

.app {
  font-family: 'Josefin Sans', sans-serif;
  color: #1ebffa;
}