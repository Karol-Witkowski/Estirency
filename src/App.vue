<template>
  <div class="app">
    <Theme />
    <Navbar />
    <router-view />
  </div>
</template>

<script>
import Navbar from './components/navbar/Navbar.vue';
import Theme from './components/Theme.vue';

export default {
  components: {
    Navbar,
    Theme,
  },
  name: 'App',
  data() {
    return {
    };
  },
};
</script>

<style src="./App.scss" lang="scss" />
