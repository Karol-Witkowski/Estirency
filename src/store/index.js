import Vue from 'vue';
import Vuex from 'vuex';
import currencies from '@/data/currencies.json';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    currency: currencies,
    baseCurrency: currencies[21],
    targetCurrency: currencies[9],
    amountValue: 100,
  },
  mutations: {
    setBaseCurrency(state, base) {
      state.baseCurrency = base;
    },
    setTargetCurrency(state, wanted) {
      state.targetCurrency = wanted;
    },
    setAmountValue(state, amount) {
      state.amountValue = amount;
    },
  },
});
