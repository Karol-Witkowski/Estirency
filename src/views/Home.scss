.home {
  display: flex;
  box-sizing: border-box;
}