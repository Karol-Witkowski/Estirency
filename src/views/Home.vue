<template>
  <div>
    <SelectMenu />
    <Rate data-cy="rate" />
    <ChartComponent />
  </div>
</template>

<script>
import ChartComponent from '@/components/ChartComponent.vue';
import SelectMenu from '@/components/SelectMenu.vue';
import Rate from '@/components/Rate.vue';

export default {
  name: 'Home',
  components: {
    ChartComponent,
    SelectMenu,
    Rate,
  },
};
</script>

<style src="@/stylesheets/Home.scss" lang="scss" scoped />
