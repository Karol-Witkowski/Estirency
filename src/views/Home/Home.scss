.home {
  display: flex;
  box-sizing: border-box;
  @media only screen and (max-width: 950px) {
    flex-wrap: wrap;
  }
}