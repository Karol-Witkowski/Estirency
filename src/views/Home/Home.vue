<template>
  <main>
    <SelectMenu class="selectMenu"/>
    <Rate class="rate"/>
    <ChartComponent class="chart"/>
  </main>
</template>

<script>
import ChartComponent from '@/components/chartComponent/ChartComponent.vue';
import SelectMenu from '@/components/selectmenu/SelectMenu.vue';
import Rate from '@/components/rate/Rate.vue';

export default {
  name: 'Home',
  components: {
    ChartComponent,
    SelectMenu,
    Rate,
  },
};
</script>

<style src="./Home.scss" scoped lang="scss"/>
