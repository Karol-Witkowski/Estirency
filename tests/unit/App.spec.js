import { expect } from 'chai';
import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vue from 'vue';
import VueRouter from 'vue-router';
import App from '@/App.vue';
import Home from '@/views/Home/Home.vue';
import store from '@/store';

const localVue = createLocalVue();
localVue.use(VueRouter);

Vue.use(VueRouter);

describe('App', () => {
  it('Check the app name and components existence', () => {
    const wrapper = shallowMount(App);

    expect(wrapper.name()).to.match(/App/);
    expect(wrapper.findAll('.theme').exists());
    expect(wrapper.findAll('.navigation').exists());
    expect(wrapper.findAll('.router-view').exists());
  });

  it('renders a child component via routing', async () => {
    const router = new VueRouter({
      routes: [
        { path: '/home', name: 'Home', component: Home },
        { path: '/about', name: 'About', component: { render: (h) => h } },
      ],
    });
    const wrapper = mount(App, {
      localVue,
      router,
      store,
    });

    router.push('/home');
    await wrapper.vm.$nextTick();

    expect(wrapper.find(Home).exists()).to.equal(true);
  });

  it('should have a different route than /home', async () => {
    const router = new VueRouter({
      routes: [
        { path: '/home', name: 'Home', component: Home },
        { path: '/about', name: 'About', component: { render: (h) => h } },
      ],
      mode: 'abstract',
    });
    const wrapper = mount(App, {
      localVue,
      router,
    });

    expect(wrapper.find(Home).exists()).to.equal(false);
  });
});
