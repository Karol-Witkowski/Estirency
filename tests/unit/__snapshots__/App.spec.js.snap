// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue test Render correctly 1`] = `
<main>
  <div></div>
  <nav class="navbar"><a href="#/home" class="">
      <div class="logo"><img alt="bell curve chart icon" src="@/assets/chart-bell-curve.png"> <span>esti</span> <span>rency</span></div>
    </a>
    <nav class="effect">
      about
    </nav>
    <nav class="effect">
      home
    </nav>
  </nav>
  <!---->
</main>
`;
